### PR TITLE
Add itertools to stdx

### PR DESCRIPTION
**Pull request by alisha17**
Added example for itertools crate

Fix to issue #15

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -303,6 +304,35 @@ fn main() { run().unwrap() }
 &nbsp;&NewLine;&nbsp;&NewLine;&nbsp;&NewLine;
 
 
+<a id="itertools"></a>
+### `itertools = "0.6.0"` &emsp; [📖][d-itertools]
+
+Itertools provides the extra iterator adaptors, functions and
+macros.
```
Comment (brson) [inline README.md]: The word "the" here is extraneous.

I think we can say tell a more interesting story in this description. Something like

"The Rust standard `Iterator` (hyperlinked) type provides a powerful abstraction for operating over sequences of values,
and is used pervasively throughout Rust. There are though a number of common operations one might want to perform on sequences that are not provided by the standard library, and that's where `itertools` comes in. If you've ever found yourself missing ... etc."

stdx actually used to contain itertools, and [here's the text it used to use](https://github.com/brson/stdx/commit/eb0d625f5d1ed87936e74a04063bcc8368882fc4#diff-04c6e90faac2675aa89e2176d2eec7d8L114), which is suitably colorful.

File: examples/itertools.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+let a = [3, 2, 5, 8, 7];
+
+// Combine all iterator elements into one String, seperated by *.
+println!("{:?}", join(&a, "*"));
+// Return the maximum value of the iterable.
+println!("{:?}", max(a.iter()).unwrap());
+// Collect all the iterable's elements into a sorted vector in
+// ascending order.
+println!("{:?}", sorted(a.iter()));
+}
```
Comment (brson) [inline examples/itertools.rs]: Indentation again.

**Comment by brson** [comment]
Thanks again @alisha17!

**Comment by brson** [comment]
Ping @alisha17 updates?

**Comment by alisha17** [comment]
@brson Sorry, I was stuck in some work these days. Will update it today itself.

**Comment by brson** [comment]
Thanks @alisha17 !

**Pull request merged**